src/assets/scripts/client/base/BaseModel.js

```diff
@@ -45,4 +49,22 @@ export default class BaseModel {
     reset() {
         throw new TypeError('BaseModel#reset method must be implemented by the class extending BaseModel');
     }
+
+    /**
+     * This will verify if the given argument is a string otherwise it will return 'Base Model'
+     *
+     * @for BaseModel
+     * @method veriftyModelName
+     * @param {string / Object}
+     */
+    buildIdPrefix(modelName) {
+        if (!_isString(modelName)) {
+            console.error('BaseModel#constructor expects a string for its first paramert but a string was not given');
+            return 'BaseModel-';
```

Comment:
Also if I do a `throw` then it will never hit the return statement and things will break? (not sure on the term obtrusive error handling?) Or should this be non-obtrusive and just give a default response?